File: backend/app/schemas/risk.py
from pydantic import BaseModel


class RiskRequest(BaseModel):

    application_id: int

    income: float

    credit_score: int

    loan_amount: float

    loan_term: int

    employment_years: float

    digital_activity_score: float = 0.0

    transaction_consistency: float = 0.0


class RiskResponse(BaseModel):

    application_id: int

    risk_score: float

    risk_level: str

    decision: str

File: frontend/src/api/risk.js
import client from "./client";

/**
 * POST /risk/score
 * body: RiskRequest {
 *   application_id, income, credit_score, loan_amount, loan_term,
 *   employment_years, digital_activity_score, transaction_consistency
 * }
 * returns: RiskResponse { application_id, risk_score, risk_level, decision }
 */
export function scoreRisk(data) {
  return client.post("/risk/score", data).then((res) => res.data);
}
